File: src/gm/template/python/types/tests/testCompositeType.py
import unittest
import gm

class Test{{ valueType.className }}(unittest.TestCase):

    def testDefaultInitialization(self):
        gm.{{ valueType.className }}()

    def testElementInitialization(self):
        gm.{{ valueType.className }}(
{%- for index in range(valueType.elementSize) -%}
            gm.{{ valueType.elements[ index ].type.className }}()
{%- if index + 1 < valueType.elementSize -%}
        ,{{ " " }}
{%- endif -%}
{%- endfor -%}
        )

File: src/gm/template/types/tests/testCompositeType.cpp
#include <catch2/catch.hpp>

#include <gm/types/{{ valueType.headerFileName }}>

TEST_CASE( "{{ valueType.className }}_DefaultConstructor" )
{
    gm::{{ valueType.className }} {{ valueType.varName }};
{% for element in valueType.elements -%}
    CHECK( {{ valueType.varName }}.{{ element.accessorName }}() ==
{%- if not element.type.isScalar -%}
        gm::
{%- endif -%}
           {{ element.type.className }}() );
{%- endfor %}
}
